framework/packages/abstract-client/src/account.rs

```diff
@@ -831,24 +865,24 @@ impl<Chain: CwEnv> Account<Chain> {
                 .iter()
                 .any(|module_info| module_info.id == m.module.id())
         });
+        if !modules.is_empty() {
+            self.abstr_account
+                .manager
+                .install_modules(modules, Some(funds))?;
+        }
+        let module_addresses = self.module_addresses(vec![M::module_id().to_owned()])?;
 
-        let install_module_response = self
-            .abstr_account
-            .manager
-            .install_modules(modules, Some(funds))?;
-
-        let module_addr = Self::parse_modules_installing_response(install_module_response);
         let contract = Contract::new(
             M::installed_module_contract_id(&self.id()?),
             self.environment(),
         );
-        contract.set_address(&module_addr);
+        contract.set_address(&module_addresses.modules[0].1);
```

Comment:
is this guaranteed to be index 0 on this array?

Reply:
Yeah, we error when we get no modules
https://github.com/AbstractSDK/abstract/blob/6a6abf82ca05bdefc6f21f58de31bc7ffb302452/framework/contracts/account/manager/src/queries.rs#L187-L191

Reply:
Updated this query method on manager!